Fixes #158, Added html lint

adarsh-ideal opened the pull request:
Html linter added as per issue https://github.com/fossasia/loklak_search/issues/158
scss linter was not working, there was "and" instead of "&&" operator, fixed it as well.

mariobehling commented:
Travis still failing. Please check.

adarsh-ideal commented:
Yes sir, checking that.

adarsh-ideal commented:
I'll revert them, tomorrow.

adarsh-ideal commented:
@hemantjadon Reverted them.

hemantjadon commented on src/app/contact/contact-form/contact-form.component.scss:
```diff
@@ -46,8 +46,8 @@
-    			*display: inline;
-    			*zoom: 1;
+    			display: inline;
+    			zoom: 1;
```
what this zoom property doing here?? please remove this as it is a non standard property

adarsh-ideal replied:
It was already there, I just removed the asteriks because it was giving error when I ran the tests locally.

hemantjadon commented on .htmllintrc:
```diff
@@ -0,0 +1,15 @@
+{
+ "attr-name-ignore-regex": false,
+ "attr-name-style":["lowercase"],
```
I think it should be `kebab-case`. please check.

adarsh-ideal replied:
It's not in the manual https://github.com/htmllint/htmllint/wiki/Options#attr-name-style
![attr](https://cloud.githubusercontent.com/assets/12227945/22110197/d6d243e4-de80-11e6-8fc7-b1de51ae42fd.png)

jayvdb replied:
The manual page you linked does show the format name for kebab case.

adarsh-ideal replied:
Yes that's why I didn't use kebab-case.

adarsh-ideal replied:
Actually it is there https://github.com/htmllint/htmllint/wiki/Options
but its not `kebab-case`, its `dash`

hemantjadon replied:
use dash

adarsh-ideal replied:
dash is there in my latest commit

yukiisbored commented on tslint.json:
```diff
@@ -109,3 +109,4 @@
     "directive-class-suffix": true
   }
 }
+
```
Why is this unnecessary blank line is added?
It doesn't look like a missing newline on EOL fix like https://github.com/fossasia/loklak_search/pull/208/files#diff-c0745258b9513d87ab38c11820efcdfbR95

yukiisbored commented on src/app/contact/contact-form/contact-form.component.scss~:
```diff
@@ -0,0 +1,136 @@
+.overlay {
```
What's the reason for this? The Tilda at the end of the file name usually indicates a backup file. Is this a backup file?

sudheesh001 replied:
This looks like a backup file and needs to be deleted.

adarsh-ideal commented:
@hemantjadon Replaced `lowercase` with `dash` & deleted the backup scss file.

hemantjadon commented:
Can html lint test for indentation and line length

adarsh-ideal commented:
@hemantjadon Yes, it tests for indentation & I added line length test,
git automatically changed mode of rest files (maybe because I moved to a new distro).
Please check this..
![untitled](https://cloud.githubusercontent.com/assets/12227945/22175544/b69d44b4-e01c-11e6-933b-ae9042c73c98.png)

hemantjadon commented on .htmllintrc:
```diff
@@ -0,0 +1,17 @@
+ "attr-no-dup": true,
+ "attr-quote-style":["quoted"],
+ "attr-req-value": true,
+ "indent-style": ["spaces"],
```
Make it tabs

hemantjadon commented on .htmllintrc:
```diff
@@ -0,0 +1,17 @@
+ "attr-quote-style":["quoted"],
+ "attr-req-value": true,
+ "indent-style": ["spaces"],
+ "line-max-len": ["100"],
```
140 line length

mariobehling commented:
This PR has a number of issues. Please check the outcome and re-open a PR when these have been fixed. Thank you.
